Assets/Scripts/Battlefield/CreatureScripts/UnitBase.cs

```diff
@@ -48,15 +48,12 @@ public void StartTurn()
 
         protected void MoveTo(Vector3 pos)
         {
-            if (currentTile)
-            {
-                currentTile.unitOnTile = null;
-            }
             agent.destination = pos;
         }
 
         public void SetTile(Tile tile)
         {
+            currentTile.unitOnTile = null;
             currentTile = tile;
             tile.unitOnTile = this.gameObject;
         }
```

Comment:
Use unity trigger to do this, it works really will reduce future issues. set tile in Unit Script and Unit in Tile script at once using OnTriggerEnter and onTriggerExit.